solutionbox/code_free_ml/analyze_data.py

```diff
@@ -78,7 +84,67 @@ def parse_arguments(argv):
     ValueError: for bad parameters
   """
   parser = argparse.ArgumentParser(
-      description='Runs analysis on structured data.')
+      formatter_class=argparse.RawDescriptionHelpFormatter,
+      description=textwrap.dedent("""\
+          Runs analysis on structured data and produces auxiliary files for
+          training. The output files can also be used by the Transform step
+          to materialize TF.Examples files, which for some problems can speed up
+          training.
+
+          Description of input files
+          --------------------------
+
+          1) If using csv files, the --csv-schema-file must be the file path to
+             a schema file. The format of this file must be a valid BigQuery
+             schema file, which is a JSON file containing a list of dicts.
+             Consider the example schema file below:
+
+             [
+                {"name": "column_name_1", "type": "integer"},
+                {"name": "column_name_2", "type": "float"},
+                {"name": "column_name_3", "type": "string"},
+                {"name": "column_name_4", "type": "string"},
+             ]
+
+             Note that the column names in the csv file much match the order
+             in the schema list. Also, we only support three BigQuery types (
+             integer, float, and string).
+
+             If instead of csv files, --bigquery-table is used, the schema file
+             does not have to be pass in as this program will extract it from
+             the table directly.
+
+          2) --features-file is a file path to a file describing the
+             transformations. Below is an example features file:
+
+             {
+                "column_name_1": {"transform": "scale"},
+                "column_name_3": {"transform": "target"},
+                "column_name_2": {"transform": "one_hot"},
+                "column_name_4": {"transform": "key"},
+             }
+
+             The format is of the dict is `name`: `transform-dict` where the
+             `name` must be a column name from the schema file. A list of
+             supported `transform-dict`s is below:
+
+             {"transform": "identity"}: does nothing (for numerical columns).
+             {"transform": "scale", "value": x}: scale a numerical column to
+                [-a, a]. If value is missing, x defaults to 1.
+             {"transform": "one_hot"}: makes a one-hot encoding of a string
+                column.
+             {"transform": "embedding", "embedding_dim": d}: makes an embedding
+                of a string column.
+             {"transform": "bag_of_words"}: bag of words transform for text
+                columns.
```

Comment:
Text columns are no different from string columns from their schema. Should we call them string columns?